clippy_lints/src/methods/unnecessary_map_or.rs

```diff
@@ -60,6 +60,8 @@ pub(super) fn check<'a>(
         Some(_) | None => return,
     };
 
+    let ext_def_span = def.span.with_hi(map.span.lo());
```

Comment:
```suggestion
    let ext_def_span = def.span.until(map.span);
```
Simpler and also correctly handles spans from different contexts (although it doesn't really make a difference for this lint as the methods pass already filters out calls where arguments are macro calls)

Reply:
Done